src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -984,10 +984,11 @@ def _validate_vm_vmss_create_auth(namespace, cmd=None):
 
     if namespace.admin_username is None:
         namespace.admin_username = get_default_admin_username()
-    namespace.admin_username = _validate_admin_username(namespace.admin_username, namespace.os_type)
+    if namespace.admin_username and namespace.os_type:
+        namespace.admin_username = _validate_admin_username(namespace.admin_username, namespace.os_type)
 
-    if not namespace.os_type:
-        raise CLIError("Unable to resolve OS type. Specify '--os-type' argument.")
+    # if not namespace.os_type:
+    #     raise CLIError("Unable to resolve OS type. Specify '--os-type' argument.")
```

Comment:
May I ask why disable this verification?

Reply:
skip here but add check in the following conditions, so if it is None, most check will be skipped

Reply:
OK